boards/Makefile.common

```diff
@@ -24,7 +28,10 @@ RUSTFLAGS_FOR_CARGO ?= \
   -C linker-flavor=ld.lld \
   -C relocation-model=dynamic-no-pic \
   -C link-arg=-zmax-page-size=512 \
-  --remap-path-prefix=$(MAKEFILE_PARENT_PATH)= \
+  --remap-path-prefix=$(TOCK_ROOT_DIRECTORY)= \
+
+RUSTFLAGS_FOR_BIN ?= \
```

Comment:
What does this do?

Reply:
I added documentation, thanks for the reminder. A longer explanation is in https://github.com/tock/tock/pull/1714#issuecomment-606017599.

For now, I moved only one linker flag (which was enough to have Cargo caching builds of all dependencies), but in principle more of them could move to this. But I'd prefer to evaluate that in a separate pull-request to try to keep this one as atomic as possible and reduce the risk of breakage.